machine/v1beta1/types_azureprovider.go

```diff
@@ -313,7 +313,8 @@ type DiskSettings struct {
 type OSDiskManagedDiskParameters struct {
 	// StorageAccountType is the storage account type to use.
 	// Possible values include "Standard_LRS", "Premium_LRS".
-	StorageAccountType string `json:"storageAccountType"`
+	// +kubebuilder:validation:Enum=Standard_LRS;Premium_LRS
```

Comment:
What previous happened if a user set this field to a value other than one of these in the past?  did an operator indicate failure anywhere? Did machines get stuck?  Did some undefined behavior happen?

If we had a clear failure, there's a decent case for fixing this.  If it did *something* previously, we may have to live with the previous mistakes.

Reply:
With:
```
osDisk:
  diskSizeGB: 128
  managedDisk:
    storageAccountType: Foo
  osType: Linux
```
Previously the Machine would go into `Failed` state.

A describe on it would report:
```
Status:
  Conditions:
    Last Transition Time:  2022-03-14T18:50:56Z
    Status:                True
    Type:                  Drainable
    Last Transition Time:  2022-03-14T18:50:56Z
    Message:               Instance has not been created
    Reason:                InstanceNotCreated
    Severity:              Warning
    Status:                False
    Type:                  InstanceExists
    Last Transition Time:  2022-03-14T18:50:56Z
    Status:                True
    Type:                  Terminable
  Error Message:           failed to reconcile machine "ddonati-test57-zj955-worker-eastus3-3-d6wjp": failed to create vm ddonati-test57-zj955-worker-eastus3-3-d6wjp: failure sending request for machine ddonati-test57-zj955-worker-eastus3-3-d6wjp: cannot create vm: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="Error converting value \"Foo\" to type 'System.Nullable`1[Microsoft.Azure.XStoreClientHelper.StorageAccountServiceClient.StorageAccountType]'. Path 'properties.storageProfile.osDisk.managedDisk.storageAccountType', line 1, position 4016." Target="vm.properties.storageProfile.osDisk.managedDisk.storageAccountType"
  Error Reason:            InvalidConfiguration
  Last Updated:            2022-03-14T18:50:59Z
  Phase:                   Failed
  Provider Status:
    Conditions:
      Last Probe Time:       2022-03-14T18:50:59Z
      Last Transition Time:  2022-03-14T18:50:59Z
      Message:               failed to create vm ddonati-test57-zj955-worker-eastus3-3-d6wjp: failure sending request for machine ddonati-test57-zj955-worker-eastus3-3-d6wjp: cannot create vm: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="Error converting value \"Foo\" to type 'System.Nullable`1[Microsoft.Azure.XStoreClientHelper.StorageAccountServiceClient.StorageAccountType]'. Path 'properties.storageProfile.osDisk.managedDisk.storageAccountType', line 1, position 4016." Target="vm.properties.storageProfile.osDisk.managedDisk.storageAccountType"
      Reason:                MachineCreationFailed
      Status:                True
      Type:                  MachineCreated
    Metadata:
Events:
  Type     Reason        Age   From              Message
  ----     ------        ----  ----              -------
  Warning  FailedCreate  12s   azure-controller  InvalidConfiguration: failed to reconcile machine "ddonati-test57-zj955-worker-eastus3-3-d6wjp": failed to create vm ddonati-test57-zj955-worker-eastus3-3-d6wjp: failure sending request for machine ddonati-test57-zj955-worker-eastus3-3-d6wjp: cannot create vm: compute.VirtualMachinesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="Error converting value \"Foo\" to type 'System.Nullable`1[Microsoft.Azure.XStoreClientHelper.StorageAccountServiceClient.StorageAccountType]'. Path 'properties.storageProfile.osDisk.managedDisk.storageAccountType', line 1, position 4016." Target="vm.properties.storageProfile.osDisk.managedDisk.storageAccountType"
```

Reply:
Ok, but the failure wasn't instant.  This is an unfortunate state to be in, but I don't think we can correct/fix the validation until we have an easy way to express racheting validation.